AddressSpace/src/FreeVariablesEngine.cpp

```diff
@@ -0,0 +1,211 @@
+            if (initialValueAsString != "1" && initialValueAsString != "0") // booleans are from discrete set.
+            {
+                throw_runtime_error_with_origin("While instantiating FreeVariable '"
+                        + config.name() + "' under '"
+                        + parentNodeId.toString().toUtf8()
+                        + " boolean literal is wrong '" + initialValueAsString + "'. Allowed values are 0 or 1.");
+            }
+            value = initialValueAsString == "1" ? OpcUa_True : OpcUa_False;
+        }
+        initialValue.setBool( value );
+    }
+    else if (config.type() == "Byte")
+        initializeIntegerNumeric( initialValue, &UaVariant::setByte, &std::stoul, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "SByte")
+        initializeIntegerNumeric( initialValue, &UaVariant::setSByte, &std::stol, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "UInt16")
+        initializeIntegerNumeric( initialValue, &UaVariant::setUInt16, &std::stoul, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "Int16")
+        initializeIntegerNumeric( initialValue, &UaVariant::setInt16, &std::stol, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "UInt32")
+        initializeIntegerNumeric( initialValue, &UaVariant::setUInt32, &std::stoul, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "Int32")
+        initializeIntegerNumeric( initialValue, &UaVariant::setInt32, &std::stol, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "UInt64")
+        initializeIntegerNumeric( initialValue, &UaVariant::setUInt64, &std::stoul, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "Int64")
+        initializeIntegerNumeric( initialValue, &UaVariant::setInt64, &std::stol, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "Double")
+        initializeFloatingPointNumeric( initialValue, &UaVariant::setDouble, &std::stod, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "Float")
+        initializeFloatingPointNumeric( initialValue, &UaVariant::setFloat, &std::stof, config.initialValue().present(), config.initialValue().get() );
+    else if (config.type() == "String")
+        initialValue.setString( config.initialValue().present()? config.initialValue().get().c_str() : "");
+    else
+        throw_runtime_error_with_origin("Free variable: given type seems invalid: '"+config.type()+"'");
+
+    // if declared as numeric data type, make it available as input for calculated variables.
+    if (config.type() == "Byte"   || config.type() == "SByte" || config.type() == "UInt16" || config.type() == "Int16" ||
```

Comment:
Well, sounds indeed like a 'good idea' however note that we already have functions for this in the basic quasar where we apply code generations. Here we are doing things very dynamically, I think it is still quite an isolated case and it might stay isolated for long time. So I'd probably not invest that much into it.